_playwright-tests/UI/PopularRepo.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { describe } from 'node:test';
+import { test, type Page } from '@playwright/test';
+import { navigateToRepositories } from './helpers/navHelpers';
+import { deleteAllRepos } from './helpers/deleteRepositories';
+import { closePopupsIfExist } from './helpers/helpers';
+
+describe('Popular Repositories', () => {
+  test('Clean - Delete any current repos that exist', async ({ page }) => {
+    await deleteAllRepos(page);
+  });
+
+  test('Navigate to Popular Repos tab, assert the Add button is displayed', async ({ page }) => {
+    await navigateToRepositories(page);
+    await closePopupsIfExist(page);
+
+    const AddMenuButton = page.locator('[data-ouia-component-id="add_checked_repos-with-snap"]');
+    await page.getByRole('link', { name: 'Popular repositories' }).click();
+
+
+    // Wait for the Add button to become enabled (up to 10 seconds)
+    await AddMenuButton.first().isEnabled({ timeout: 10000 });
+  });
+
+  test('Add popular repos', async ({ page }) => {
+    await navigateToRepositories(page);
+    await closePopupsIfExist(page);
+
+    const AddMenuButton = page.locator('[data-ouia-component-id="add_popular_repo_toggle"]');
+    const AddButton = page.locator('[data-ouia-component-id="add_popular_repo"]');
```

Comment:
This is used exactly once - maybe just use it inline?

Reply:
This is partly habit, we currently define buttons separately from the test. I would prefer the test to be consistent (either define above or inline rather than mix methods).

Reply:
with https://github.com/content-services/content-sources-frontend/pull/428  you can use  page.getByTestId('add_popular_repo')    lets get that merged!

Reply:
just merged so that's an option, you'll just need to rebase

Reply:
Done that, thank you